src/material/dialog/testing/dialog-harness.ts

```diff
@@ -11,56 +11,57 @@ import {DialogRole} from '@angular/material/dialog';
 import {DialogHarnessFilters} from './dialog-harness-filters';
 
 /**
- * Harness for interacting with a standard MatDialog in tests.
+ * Harness for interacting with a standard `MatDialog` in tests.
  * @dynamic
  */
 export class MatDialogHarness extends ComponentHarness {
   // Developers can provide a custom component or template for the
   // dialog. The canonical dialog parent is the "MatDialogContainer".
+  /** The selector for the host element of a `MatDialog` instance. */
   static hostSelector = '.mat-dialog-container';
 
   /**
-   * Gets a `HarnessPredicate` that can be used to search for a dialog with
-   * specific attributes.
-   * @param options Options for narrowing the search:
-   *   - `id` finds a dialog with specific id.
+   * Gets a `HarnessPredicate` that can be used to search for a `MatDialogHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which dialog instances are considered a match.
    * @return a `HarnessPredicate` configured with the given options.
    */
   static with(options: DialogHarnessFilters = {}): HarnessPredicate<MatDialogHarness> {
     return new HarnessPredicate(MatDialogHarness, options);
   }
 
-  /** Gets the id of the dialog. */
+  /** Gets a promise for the id of the dialog. */
   async getId(): Promise<string|null> {
```

Comment:
I'm somehow (personally) not happy about the way we need to repeat `promise for` all the time 😞 I've been following the concept of just saying e.g. `Gets the text of the tab content` The JSDoc method description does not necessarily refer to return value _type_.

I think we need to decide on how we want to document such methods in general. i.e. if we want to mention `Promise` each time, and what consistent pattern we should follow (like `Gets a promise for xxx`).

As an example: https://jsdoc.app/tags-returns.html. The function w/ promise does not mention promise in the function description. (`@returns` is equal to the type in TS)

Reply:
I'm not trying to block this PR going forward. I just thought about this too and would love to discuss this at some point (maybe we could even decide on a consistent format for future asynchronous methods)

Reply:
I could also see just not mentioning the fact that its a promise, but we should be consistent about it. I noticed that prior to this PR it was a mix of comments that mention it and ones that don't, so I chose to change them all to mention it.

Reply:
Yes. agreed. At the moment it is inconsistent, so this PR is definitely good! We can revisit this in the future. I'd prefer not mentioning promise, but we can discuss this together and made changes based on the outcome.

Reply:
This seems like a decent bikeshed to paint during the team meeting.

Reply:
(my opinion would be to just include the word "asynchronously" in the description, e.g. "Get the name of the button asynchronously").